### PR TITLE
Add MigrationZapper

### PR DESCRIPTION
**Pull request by shahthepro**
Right now, it's impossible to convert OGV to OGN and stake it in a single call if you don't have any OGV lockups with our Migrator contract.

The new `MigrationZapper` will allow users to do that. It internally calls `Migrator.migrate` and then stakes it on xOGN contract.

The simple `migrate(uint256)` should:
- Transfer in OGV from user's wallet
- Migrate it to OGN
- And transfer OGN back to the user

The other `migrate(uint256,uint256,uint256)` should:
- Transfer in OGV from user's wallet
- Migrate it to OGN
- Stake it on behalf of the user for specified duration
- Transfer back any leftover OGN to user
---------
If you made a contract change, make sure to complete the checklist below before merging it in master.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. 
  - [ ] Copy & paste code review [security checklist](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md) below this checklist.
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/MigrationZapper.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     *         and stakes it.
+     * @param ogvAmount Amount of OGV to migrate
+     */
+    function migrate(uint256 ogvAmount, uint256 newStakeAmount, uint256 newStakeDuration) external {
+        // Take tokens in
+        ogv.transferFrom(msg.sender, address(this), ogvAmount);
+
+        // Migrate
+        uint256 ognReceived = migrator.migrate(ogvAmount);
+
```
Comment (sparrowDom) [inline contracts/MigrationZapper.sol:60]: we could add a: 
```
require(ognReceived >= newStakeAmount, "Not enough OGN received");
```

Afaik ogn/ogv conversion rates should be locked, and such check should only be triggered by faulty math. Though it is still nice to have an understandable error message

Comment (shahthepro) [reply contracts/MigrationZapper.sol:60]: Good point, we were just running into this error (DApp was sending a higher amount due to a rounding issue)

**Review by sparrowDom — APPROVED** [review]
LGTM

File: contracts/MigrationZapper.sol [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    constructor(address _ogv, address _ogn, address _migrator, address _ognStaking, address _governor) {
+        ogv = IMintableERC20(_ogv);
+        ogn = IMintableERC20(_ogn);
+        migrator = IMigrator(_migrator);
+        ognStaking = IStaking(_ognStaking);
+
+        governor = _governor;
+    }
+
+    function initialize() external {
```
Comment (DanielVF) [inline contracts/MigrationZapper.sol:28]: Fortuity, `initialize` can't do any damage, so it's okay if it's called multiple times, but this should have an initializer modifier.

File: contracts/MigrationZapper.sol [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+     * Transfers any tokens sent by mistake out of the contract
+     * @param token Token address
+     * @param amount Amount of token to transfer
+     */
+    function transferTokens(address token, uint256 amount) external {
+        if (msg.sender != governor) {
+            revert NotGovernor();
+        }
+
+        IMintableERC20(token).transfer(governor, amount);
```
Comment (DanielVF) [inline contracts/MigrationZapper.sol:86]: Nobody should send us USDC, but a generic token transfer method should usually use safeMath.

File: contracts/MigrationZapper.sol (hunk 2 above)
Comment (DanielVF) [reply contracts/MigrationZapper.sol:28]: Alternatively, if no proxy, this initialize code could move into constructor.

**Pull request merged**